### ISSUE TITLE
Feature request: maintain state between value types

### ISSUE THREAD

**Issue by tcullum-rh** [dominikwilkowski/vault#95]
As a user, when adding a new field to a record, I would like for the field state to remain when going from a Single Line or Secret Single Line to a Multi-Line or Secret Multi-Line or Heading.

### Description

Currently, the default data type for a new field in a record is Single Line Secret. This is good. However, if I enter a value, such as "test" in there, and I subsequently switch to Multi-Line, Secret Multi-Line, or Heading, the current value in the field seems to be erased and replaced with a boiler plate text of "Value of field".

### Proposed Change

If there is already text in the value field, it remains when the value type is changed from Single Line or Secret Single Line to Multi-Line, Secret Multi-Line and Heading.

### Notes

The field text already maintains state when going from  either Single Line type to URL or vice versa.

**Comment by dominikwilkowski** [comment]
Yeah I noticed that too and like this idea. Let me see what I can do in the next few days.
